org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/util/SelectionUtils.java

```diff
@@ -45,4 +55,74 @@ public static Object getSingleElement(ISelection s) {
     return selection.getFirstElement();
   }
 
+  public static Set<ISonarLintProject> allSelectedProjects(ExecutionEvent event) throws ExecutionException {
+    ISelection selection = HandlerUtil.getCurrentSelectionChecked(event);
+
+    final Set<ISonarLintProject> selectedProjects = new HashSet<>();
+
+    if (selection instanceof IStructuredSelection) {
+      List elems = ((IStructuredSelection) selection).toList();
+      for (Object elem : elems) {
+        if (elem instanceof IAdaptable) {
+          collectProjects(selectedProjects, (IAdaptable) elem);
+        }
+      }
+    }
+
+    return selectedProjects;
+
+  }
+
+  private static void collectProjects(Set<ISonarLintProject> selectedProjects, IAdaptable elem) {
+    ISonarLintProjectContainer container = (ISonarLintProjectContainer) elem.getAdapter(ISonarLintProjectContainer.class);
+    if (container != null) {
+      selectedProjects.addAll(container.projects());
+      return;
+    }
+    ISonarLintProject project = (ISonarLintProject) elem.getAdapter(ISonarLintProject.class);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "ISonarLintProject". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1905)